plugin.picture.googlephotos/main.py

```diff
@@ -0,0 +1,395 @@
+        items.append((url, li, True))  # (url, listitem[, isFolder])
+
+    xbmcplugin.addDirectoryItems(addon_handle, items)
+    xbmcplugin.endOfDirectory(addon_handle)
+
+
+def refresh():
+    '''
+    Refreshes the current window
+    IMP: Pass previous query string as argument in the url with key prev_q
+    '''
+    id = xbmcgui.getCurrentWindowId()
+    pos = int(xbmc.getInfoLabel(f'Container.CurrentItem'))
+    cont_path = utils.build_url(
+        base_url, {'call_type': 1}, args.get('prev_q')[0])
+    xbmc.executebuiltin(f'Container.Update({cont_path})')
+
+    # get active window
+    win = xbmcgui.Window(id)
+    cid = win.getFocusId()
+    # Check if window is fully loaded
+    while not xbmc.getCondVisibility(f'Control.IsVisible({cid})'):
+        xbmc.sleep(150)
+    # Focus on the last focused position
+    xbmc.executebuiltin(f'SetFocus({cid},{pos},absolute)')
+
+
+def get_items(pageToken=None) -> dict:
+
+    # Prepare request
+    params = {'pageSize': 100}
+    if pageToken:
+        params['pageToken'] = pageToken
+    headers = {'Authorization': 'Bearer ' + token}
+
+    # Check type of required listing
+    list_type = args.get('type')[0]
+    if list_type == 'all':
+        error = __addon__.getLocalizedString(30408)
+        res = requests.get(config.service_endpoint + '/mediaItems',
+                           headers=headers, params=params)
+    elif list_type == 'album':
+        error = __addon__.getLocalizedString(30409)
+        params['albumId'] = args.get('id')[0]
+        res = requests.post(config.service_endpoint + '/mediaItems:search',
+                            headers=headers, data=params)
+    elif list_type == 'filter':
+        # params
+        params['filters'] = utils.buildFilter(__addon__)
+        if not bool(params['filters']):
+            return None
+        error = __addon__.getLocalizedString(30410)
+        res = requests.post(config.service_endpoint + '/mediaItems:search',
+                            headers=headers, json=params)
+    if res.status_code != 200:
+        dialog = xbmcgui.Dialog()
+        dialog.notification(
+            __addon__.getLocalizedString(30411), f'{error}. {__addon__.getLocalizedString(30412)}:{res.status_code}', xbmcgui.NOTIFICATION_ERROR, 3000)
+        return None
+    return res.json()
+
+
+def get_items_bg(result, path):
+    if 'nextPageToken' in result[-1]:
+        pageToken = result[-1]["nextPageToken"]
+        media = get_items(pageToken)
+        if not media:
+            return None
+        utils.storeData(path, media)
+
+
+def list_media():
+    # For all photo directories
+
+    path = profile_path + config.media_filename
+    if not args.get('call_type'):
+        xbmcvfs.delete(path)
+
+    result = utils.loadData(path)
+    if not result:
+        media = get_items()
+        if media:
+            utils.storeData(path, media)
+            result = [media]
+        else:
+            xbmc.executebuiltin(
```

Comment:
Use `xbmcgui` methods instead of `xbmc.executebuilti()`

Reply:
Done